modules/consensus/synchronize.go

```diff
@@ -247,12 +256,16 @@ func (cs *ConsensusSet) relayBlock(conn modules.PeerConn) error {
 		return err
 	}
 
-	// Submit the block to the consensus set.
+	// Submit the block to the consensus set and broadcast it.
 	err = cs.AcceptBlock(b)
 	if err == errOrphan {
 		// If the block is an orphan, try to find the parents. The block
 		// received from the peer is discarded and will be downloaded again if
 		// the parent is found.
+		//
+		// TODO: log error returned if non-nill? OR we could not use a goroutine and
+		// just return the result of the RPC call. Is it back practice to call an RPC
+		// from inside another RPC?
 		go cs.gateway.RPC(modules.NetAddress(conn.RemoteAddr().String()), "SendBlocks", cs.threadedReceiveBlocks)
```

Comment:
It becomes problematic if cyclical RPC calls are possible. In fact this chain of RPC calls is cyclical.

Also it prevents us from setting reasonable timeouts. If RPC "a" sets a timeout of 5 minutes, and RPC "b" calls RPC "a", there is no way for RPC "b" to know that it should extend its timeout by 5 minutes.

So I'll keep them called in goroutines for now and add logging later.